### PR TITLE
Adding support for Orvibo devices

### PR DESCRIPTION
**Pull request by azamat5991**

### CODE REVIEW

File: devices.js [hunk 1]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        },
+    },
+    {
+        zigbeeModel: ['095db3379e414477ba6c2f7e0c6aa026'],
+        model: 'T21W1Z',
+        vendor: 'ORVIBO',
+        description: 'Smart light switch - 1 gang',
+        supports: 'on/off',
+        fromZigbee: [fz.on_off],
+        toZigbee: [tz.on_off],
```
Comment (Koenkk) [inline devices.js:8992]: I also would expect a `configure` here like `T21W2Z` but only for endpoint 1

File: devices.js [hunk 2]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+    },
+    {
+        zigbeeModel: ['093199ff04984948b4c78167c8e7f47e'],
+        model: 'W40CZ',
+        vendor: 'Orvibo',
+        description: 'Orvibo smart curtain motor ',
+        supports: 'open, close, stop, position',
+        fromZigbee: [fz.ZNCLDJ11LM_ZNCLDJ12LM_curtain_analog_output, fz.cover_position_tilt, fz.ignore_basic_report],
```
Comment (Koenkk) [inline devices.js:9000]: Is `ZNCLDJ11LM_ZNCLDJ12LM_curtain_analog_output` really used? I wouldn't expect it.

File: devices.js [hunk 3]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        toZigbee: [tz.cover_state, tz.cover_position_tilt],
+        onEvent: async (type, data, device) => {
+            if (data.type === 'attributeReport' && data.cluster === 'genAnalogOutput') {
+                await device.endpoints[0].read('genAnalogOutput', ['presentValue']);
```
Comment (Koenkk) [inline devices.js:9004]: Why is this needed?

File: devices.js [hunk 4]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        model: 'T21W2Z',
+        vendor: 'ORVIBO',
+        description: 'Smart light switch - 2 gang',
+        supports: 'on/off',
+        fromZigbee: [fz.on_off, fz.ignore_basic_report],
+        toZigbee: [tz.on_off],
+        endpoint: (device) => {
+            return {'top': 1, 'bottom': 2};
+        },
+        meta: {configureKey: 1},
```
Comment (Koenkk) [inline devices.js:8979]: `multiEndpoint: true` is missing

File: devices.js [hunk 5]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        model: 'R11W2Z',
+        vendor: 'ORVIBO',
+        description: 'In wall switch - 2 gang',
+        supports: 'on/off',
+        fromZigbee: [fz.on_off],
+        toZigbee: [tz.on_off],
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 2};
+        },
+        meta: {configureKey: 1},
```
Comment (Koenkk) [inline devices.js:9019]: `multiEndpoint: true` is missing

File: devices.js [hunk 6]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        configure: async (device, coordinatorEndpoint) => {
+            await bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
+            await bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
+        },
+    },
+    {
+        zigbeeModel: ['9ea4d5d8778d4f7089ac06a3969e784b'],
+        model: 'R20W2Z',
+        vendor: 'ORVIBO',
+        description: 'In wall switch single Live',
```
Comment (Koenkk) [inline devices.js:9029]: Why is this not named `In wall switch - 1 gang`? (to be consistent with the `R11W2Z`)

File: devices.js [hunk 7]
```diff
@@ -8904,6 +8904,159 @@ const devices = [
...
+        fromZigbee: [fz.ias_water_leak_alarm_1],
+        toZigbee: [],
+    },
+    {
+        zigbeeModel: ['72bd56c539ca4c7fba73a9be0ae0d19f'],
+        model: 'SE21',
+        vendor: 'ORVIBO',
+        description: 'Smart emergency button',
+        supports: 'action',
+        fromZigbee: [fz.SE21_action],
```
Comment (Koenkk) [inline devices.js:9057]: I don't have access to your repo, please add the following converter to `fromZigbee.js`:

```js
    SE21_action: {
        cluster: 'ssIasZone',
        type: 'commandStatusChangeNotification',
        convert: (model, msg, publish, options, meta) => {
            const buttonStates = {
                0: 'off',
                1: 'single',
                2: 'double',
                3: 'hold',
            };

            return {action: buttonStates[msg.data.zonestatus]};
        },
    },
```

File: devices.js (hunk 1 above)
Comment (azamat5991) [reply devices.js:8992]: Most of the 1 gang switches doesn't have `configure ` in `devices.js`. Anyway, should I add something like this in addition to above code?
```
meta: {configureKey: 1},
        configure: async (device, coordinatorEndpoint) => {
            const endpoint = device.getEndpoint(1);
            await bind(endpoint, coordinatorEndpoint, ['genOnOff']);
            await configureReporting.onOff(endpoint);
        },
```

File: devices.js (hunk 6 above)
Comment (azamat5991) [reply devices.js:9029]: This is what was written on the description of the device, and actually it's not 1gang but 2gang switch. I think would be better to change to `In wall switch - 2 gang`

File: devices.js (hunk 3 above)
Comment (azamat5991) [reply devices.js:9004]: This part can be removed 
```
onEvent: async (type, data, device) => {
            if (data.type === 'attributeReport' && data.cluster === 'genAnalogOutput') {
                await device.endpoints[0].read('genAnalogOutput', ['presentValue']);
```

File: devices.js (hunk 4 above)
Comment (azamat5991) [reply devices.js:8979]: Ok, I will add it

File: devices.js (hunk 7 above)
Comment (azamat5991) [reply devices.js]: Ok

File: devices.js (hunk 1 above)
Comment (Koenkk) [reply devices.js:8992]: Yes, first test if you get a `state` update when togelling the switch via the button

Comment (azamat5991) [reply devices.js:8992]: I've tested and received a `state `update
```
zigbee2mqtt:info  2020-05-15 18:12:15: MQTT publish: topic 'zigbee2mqtt/switch_one_gang', payload '{"state":"ON","linkquality":92}'
zigbee2mqtt:debug 2020-05-15 18:12:16: Received Zigbee message from 'switch_one_gang', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID 0
zigbee2mqtt:info  2020-05-15 18:12:16: MQTT publish: topic 'zigbee2mqtt/switch_one_gang', payload '{"state":"OFF","linkquality":84}
```'

Comment (Koenkk) [reply devices.js:8992]: Then it's not needed, thanks!

Comment (azamat5991) [reply devices.js:8992]: I am receiving state update with and without the above mentioned code. So, I don’t know if it’s making any difference

**Comment by Koenkk** [comment]
Thanks!

- I made some final changes: https://github.com/Koenkk/zigbee-herdsman-converters/pull/1230/commits/36560d65d11c0c499239a8ece9ce609a173d6909 , hoping these are ok
- For my understanding, what is the difference between R11W2Z and R20W2Z ?

**Comment by azamat5991** [comment]
> Thanks!
> 
> * I made some final changes: [36560d6](https://github.com/Koenkk/zigbee-herdsman-converters/commit/36560d65d11c0c499239a8ece9ce609a173d6909) , hoping these are ok
> * For my understanding, what is the difference between R11W2Z and R20W2Z ?

It's the newest version of their No Neutral Switches, the only difference is `zigbeeModel`
